libbeat/licenser/license.go

```diff
@@ -0,0 +1,87 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package licenser
+
+import uuid "github.com/satori/go.uuid"
+
+// License represents the license of this beat, the license is fetched and returned from
+// the elasticsearch cluster.
+type License struct {
+	UUID     uuid.UUID   `json:"uid"`
+	Type     LicenseType `json:"type"`
+	Mode     LicenseType `json:"mode"`
+	Status   State       `json:"status"`
+	Features features    `json:"features"`
+}
+
+// Features defines the list of features exposed by the elasticsearch cluster.
+type features struct {
+	Graph      graph      `json:"graph"`
+	Logstash   logstash   `json:"logstash"`
+	ML         ml         `json:"ml"`
+	Monitoring monitoring `json:"monitoring"`
+	Rollup     rollup     `json:"rollup"`
+	Security   security   `json:"security"`
+	Watcher    watcher    `json:"watcher"`
```

Comment:
Should we track in here all the features or only the ones that Beats actually uses / requires?

Reply:
I've exposed the current features, maybe Graph is the less relevant for us, but we can revisit that when a new feature is exposed via elasticsearch.